client/client.go

```diff
@@ -97,7 +97,18 @@ func (ar *ApiRequest) Do(req *http.Request) (*http.Response, error) {
 	if req.Header.Get("Authorization") == "" {
 		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", ar.auth))
 	}
-	return ar.api.Do(req)
+	return logRequestIDWrapper(ar.api.Do(req))
+}
+
+// logRequestIDWrapper logs the request-id in case of an invalid response to the API
+func logRequestIDWrapper(r *http.Response, err error) (*http.Response, error) {
+	if err == nil {
+		rid := r.Header.Get("x-men-requestid")
+		if rid != "" && r.StatusCode != http.StatusOK {
+			log.Infof("requestid: %s, retCode: %d, request-url: %s", rid, r.StatusCode, r.Request.URL)
```

Comment:
@maciejmrowiec is this an appropriate amount of logging info?

Reply:
What about error messages from server? Are they logged somewhere?

Any error resp, from any of our API always contains this:
https://docs.mender.io/1.3/apis/device-apis/device-authentication#error

Reply:
http.StatusOK thats 200, everything in rage 200-399 are successful calls, 400-499 are client side errors, 500-599 server side errors

https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

Reply:
it does not look like the server error messages are logged anywhere, but I figured u would use the request id to simply read that out of the server logs serverside, or do you want them to show up in the client logs as well?

and so you want only client and server side errors to log the request-id?

Reply:
In general yes, but it would good to have context what something was happening something like: `update 123: starting postinstall scripts: executing script 123.sh: reporting update progress status: request failed with message: internal server error, request_id 123 `

Then you have enough context when you are failing, what is failing, you can use request_id to get missing pieces of the puzzle from server side. 

Probably @pasinskim would be better to review this as I have very limited understanding of client logic / logging structure.